samcli/lib/providers/sam_stack_provider.py

```diff
@@ -39,12 +44,17 @@ def __init__(
         :param dict template_dict: SAM Template as a dictionary
         :param dict parameter_overrides: Optional dictionary of values for SAM template parameters that might want
             to get substituted within the template
+        :param dict pseudo_parameter_overrides: Optional dictionary of values for SAM template pseudo parameters
+            that might want to get substituted within the template and its child templates
```

Comment:
The difference between these 2 params is, first one will only affect root stack but second one will affect all stacks (including root and its children)? If so why we don't use the first param for that purpose, since we never worked on child stacks and passed parameter overrides to the child stacks before.

Reply:
Each stack (including root and child stacks) has its own `parameter_overrides`. The `parameter_overrides` for nested stacks are defined like this:

```yaml
  SubApp:
    Type: AWS::Serverless::Application
    Properties:
      Location: ../template.yml
      Parameters:
        DefaultTimeout: ...
        MyRuntimeVersion: 'some string'
        StringValueTimeout: !Ref ABCD
```

So we cannot share `parameter_overrides` among all stacks, considering there might be parameter name clash between parent and child stacks.

Reply:
Are you trying to say we don't share parameter_overrides between nested stacks but we share pseudo_parameter_overrides? If so, I'm still unclear why not just adding(replicating) AWS::Region to each parameter_overrides

Reply:
> If so, I'm still unclear why not just adding(replicating) AWS::Region to each parameter_overrides

Reply:
> If so, I'm still unclear why not just adding(replicating) AWS::Region to each parameter_overrides

Yes that's another option. I want to make this more future proof so that in the future, if we want to pass other `AWS::*` pseudo_parameter, we can easily do so.

Reply:
Other pseudo_parameters like:

- `AWS::Partitions` ('aws' | 'aws-cn' | 'aws-us-gov')
- `AWS::URLSuffix`
- ... https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html

Right now I only find overrides for `AWS::Region` in sam-cli though.

Reply:
Now it is called `global_parameter_overrides` now, it is defintely more readable.